DataLayer/CmuData/CmuUnit.cpp

```diff
@@ -0,0 +1,60 @@
+#include "CmuUnit.h"
+
+#include <QDebug>
+
+unsigned char CmuUnit::newCmuNumber_ = 0;
+
+CmuUnit::CmuUnit()
+: cellVoltages_(QList<short>())
+, pcbTemp_(0)
+, cellTemps_(QList<unsigned short>())
+{
+    cmuNumber_ = newCmuNumber_++;
```

Comment:
Hey @enochtsang 

So I came across a fun issue with my new system of having a container for the Cmu's, Mppt's and MotorDetails... a few things

1) I had no clue how array initialisation worked anymore, correct me if I'm wrong but dynamic arrays **must** call the default constructor of the object. 
2) The QScopedArrayPointer doesn't seem to handle polymorphism as well as I'd hoped (or rather I'm doing something terribly wrong and you'll point that out to me), but it would seem the following statement gives me compile errors:
`QScopedArrayPointer<I_CmuUnit> cmuUnits_`
```
CmuData::CmuData(const unsigned char& numberOfCmus)
: cmuUnits_(new CmuUnit[numberOfCmus])
```

So this leads to an interesting problem that might influence any testing done in this area? As a solution to the first problem I have implemented a static member variable that should keep track of the cmuNumber_ for each cmuUnit (my implementation of this is in the body), but I realise that it might not even be necessary being that the CmuUnits are being stored in ordered lists... so I'd appreciate your insight on the topic. 

As for the second problem, do you see the

`QScopedArrayPointer<CmuUnit> cmuUnits_;`

I implemented causing you issues down the road, or is this alright?